### PR TITLE
add tip of w4mjoinpn and w4mcorcov; update w4mclassfilter to tip

### PR DESCRIPTION
**Pull request by eschen42**
This is a replacement for [pull request 110](https://github.com/usegalaxy-eu/usegalaxy-eu-tools/pull/110)

I would like to request the following changes:

Update w4mclassfilter to version at the repository tip
Add w4mjoinpn and w4mcorcov tools
All of these tools have detailed help and working examples.

I am preparing a manuscript to present the tools.

### CODE REVIEW

File: metabolomics.yaml.lock [hunk 1]
```diff
@@ -56,6 +56,19 @@ tools:
   owner: eschen42
   revisions:
   - 191a720488ce
+  - 582a8a42a93b
```
Comment (bgruening) [inline metabolomics.yaml.lock]: We do not need the new-old version, isn't it? Please add only the newest version.

**Comment by bgruening** [comment]
Cool thanks!

**Comment by eschen42** [comment]
To @erasche CC @bgruening @lecorguille 
So, it's great that my tools are now on UseGalaxy.eu so that I can put up the demos for my paper. :)

I do wonder, though, what determines the order in which they appear under the Metabolomics category.
I find it somewhat strange that they appear so early in the list.
I will need to discuss with the W4m team what would be an appropriate order, assuming that this is something that can be adjusted.

**Comment by bgruening** [comment]
Purely random. We can adjust the order if needed. Just let us know the wished order.

**Comment by eschen42** [comment]
@bgruening 
Thank you.  I will discuss with Gildas

**Comment by lecorguille** [comment]
One bag for all the metabolomics tools isn't fair 😝 
There are different kind of input (LC, GC, FIA, NMR, ...) and indeed some big steps (pre-precessing, normalization).
In W4M, we have plenty of room to sort the tool by workflow 😏 
(We are also thinking about ordering by steps instead because by workflow, it means to duplicate tools in the tool panel which is not easy to maintain)

@bgruening How to you manage the reordering? Manually or programmatically?

**Comment by erasche** [comment]
@lecorguille we maintain the ordering manually. We can split this out into different sections if you can tell us what you want.

**Comment by bgruening** [comment]
@lecorguille please keep in mind that we can not create so many categories as in the main server. We could do this in a subdomain as http://metagenomics.usegalaxy.eu but in the overall view we have too many other stuff as well. We could add sub-labels I think. Would this help?

**Comment by lecorguille** [comment]
I know. And that’s why I prefer dedicated instance like, I don’t know,
let’s say for example workflow4metabolomics 😛
Or indeed sub-domain.
Or workflow4metabolomics !

Seriously, it’s not easy to sort them on one level. But we can deal with
sub-label
I can propose you an organisation workflow-centric which have the advantage
to not require to duplicate tools.

Poke me if I forget


Le ven. 17 août 2018 à 18:27, Björn Grüning <notifications@github.com> a
écrit :

**Comment by bgruening** [comment]
I'm wondering how you will do this on usegalaxy.fr ...

**Comment by lecorguille** [comment]
@erasche To organise the tools within the metabolomics category.
With sub-label, I see something like that.
@eschen42 Add your own tools that aren't listed at the right place.
```
# Preprocessing-LCMS
xcms_xcmsset
xcms_merge
xcms_group
xcms_retcor
xcms_fillpeaks
camera_annotate
xcms_summary
camera_combinexsannos

# Preprocessing-FIAMS
profia

# Preprocessing-GCMS
metams_rungc

# Preprocessing-NMR
nmr_preprocessing
nmr_alignment
nmr_bucketing

# Normalization_quality-LCMS_FIAMS
batchcorrection

# Normalization_quality-ALL
normalization
transformation
qualitymetrics
withinvariation

# Statistics-ALL
univariate
multivariate
anova
biosigner
heatmap

# Annotation-LCMS_FIAMS
hmdb_ms_search
lipidmaps_textsearch
massbank_ws_searchspectrum
lcmsmatching

# Annotation-LCMS
probmetab

# Annotation-GCMS
golm_ws_lib_search

# Data_handling-ALL
generic_filter
tablemerge
checkformat
w4mclassfilter
```

**Pull request merged**